contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,971 @@
+     * @param _minTokenReceived Slippage check -> minimum amount of token expected in return
+     */
+    function rebalance(
+        uint256 _amountToSwap,
+        bool _swapWeth,
+        uint256 _minTokenReceived
+    ) external nonReentrant onlyGovernorOrStrategist {
+        /**
+         * Would be nice to check if there is any total liquidity in the pool before performing this swap
+         * but there is no easy way to do that in UniswapV3:
+         * - clPool.liquidity() -> only liquidity in the active tick
+         * - asset[1&2].balanceOf(address(clPool)) -> will include uncollected tokens of LP providers
+         *   after their liquidity position has been decreased
+         */
+
+        /**
+         * When rebalance is called for the first time there is no strategy's
+         * liquidity in the pool yet. The full liquidity removal is thus skipped.
+         */
+        if (tokenId != 0) {
+            _removeLiquidity(1e18);
+        }
+        // in some cases we will just want to add liquidity and not issue a swap to move the
+        // active trading position within the pool
+        if (_amountToSwap > 0) {
+            _swapToDesiredPosition(_amountToSwap, _minTokenReceived, _swapWeth);
+        }
+        // calling check liquidity early so we don't get unexpected errors when adding liquidity
+        // in the later stages of this function
+        _checkForExpectedPoolPrice();
+
+        _addLiquidity();
+        _checkForExpectedPoolPrice();
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /**
+     * @dev Decrease partial or all liquidity from the pool.
+     * @param _liquidityToDecrease The amount of liquidity to remove expressed in 18 decimal point
+     */
+    function _removeLiquidity(uint256 _liquidityToDecrease)
+        internal
+        gaugeUnstakeAndRestake
+    {
+        require(_liquidityToDecrease > 0, "Must remove some liquidity");
+
+        uint128 _liquidity = _getLiquidity();
+        // need to convert to uint256 since intermittent result is to big for uint128 to handle
+        uint128 _liqudityToRemove = uint256(_liquidity)
+            .mulTruncate(_liquidityToDecrease)
+            .toUint128();
+        require(_liqudityToRemove > 0, "Must remove some comp. liquidity");
+
+        (uint256 _amountWeth, uint256 _amountOethb) = positionManager
+            .decreaseLiquidity(
+                // Both expected amounts can be 0 since we don't really care if any swaps
+                // happen just before the liquidity removal.
+                INonfungiblePositionManager.DecreaseLiquidityParams({
+                    tokenId: tokenId,
+                    liquidity: _liqudityToRemove,
+                    amount0Min: 0,
+                    amount1Min: 0,
+                    deadline: block.timestamp
+                })
+            );
+
+        (
+            uint256 _amountWethCollected,
+            uint256 _amountOethbCollected
+        ) = positionManager.collect(
```

Comment:
⚪ The Gauge collects the fees before a `withdraw()`. Whenever we add or remove liquidity from the gauge, we withdraw it before doing so and deposit it back immediately after. So, this call would always collect nothing. And the returns values would always be `(0, 0)`. 

It might make the code a bit simpler and short if we can just remove this piece of code (and probably add a comment on why there's no collect here)

Reply:
See this: https://github.com/OriginProtocol/origin-dollar/pull/2184/files#r1725133557.

Apparently it is needed as the way to get the principle back out.

Reply:
Thanks, I wasn't aware of that